tools/sot_siamrpn_param_search.py

```diff
@@ -0,0 +1,272 @@
+        from mmtrack.datasets import build_dataloader
+        from mmtrack.models import build_reid as build_model
+        if 'reid' in cfg.model:
+            cfg.model = cfg.model.reid
+    else:
+        from mmtrack.apis import multi_gpu_test, single_gpu_test
+        from mmtrack.datasets import build_dataloader
+        from mmtrack.models import build_model
+    if args.cfg_options is not None:
+        cfg.merge_from_dict(args.cfg_options)
+    # set cudnn_benchmark
+    if cfg.get('cudnn_benchmark', False):
+        torch.backends.cudnn.benchmark = True
+    # cfg.model.pretrains = None
+    if hasattr(cfg.model, 'detector'):
+        cfg.model.detector.pretrained = None
+    cfg.data.test.test_mode = True
+
+    # init distributed env first, since logger depends on the dist info.
+    if args.launcher == 'none':
+        distributed = False
+    else:
+        distributed = True
+        init_dist(args.launcher, **cfg.dist_params)
+
+    # build the dataloader
+    dataset = build_dataset(cfg.data.test)
+    data_loader = build_dataloader(
+        dataset,
+        samples_per_gpu=1,
+        workers_per_gpu=cfg.data.workers_per_gpu,
+        dist=distributed,
+        shuffle=False)
+
+    logger = get_logger('ParamsSearcher', log_file=args.log)
+
+    # build the model and load checkpoint
+    if cfg.get('test_cfg', False):
+        model = build_model(
+            cfg.model, train_cfg=cfg.train_cfg, test_cfg=cfg.test_cfg)
+    else:
+        model = build_model(cfg.model)
+    fp16_cfg = cfg.get('fp16', None)
+    if fp16_cfg is not None:
+        wrap_fp16_model(model)
+    if args.checkpoint is not None:
+        checkpoint = load_checkpoint(
+            model, args.checkpoint, map_location='cpu')
+        if 'CLASSES' in checkpoint['meta']:
+            model.CLASSES = checkpoint['meta']['CLASSES']
+    if not hasattr(model, 'CLASSES'):
+        model.CLASSES = dataset.CLASSES
+
+    if args.fuse_conv_bn:
+        model = fuse_conv_bn(model)
+
+    if not distributed:
+        model = MMDataParallel(model, device_ids=[0])
+    else:
+        model = MMDistributedDataParallel(
+            model.cuda(),
+            device_ids=[torch.cuda.current_device()],
+            broadcast_buffers=False)
+
+    # search hyper parameters coarsely
+    # penalty_k_range = np.round(np.arange(0.05, 0.5, 0.05), decimals=2)
+    # lr_range = np.round(np.arange(0.35, 0.5, 0.05), decimals=2)
+    # win_influ_range = np.round(np.arange(0.1, 0.8, 0.1), decimals=2)
+
+    if args.dataset == 'otb':
+        # fine search on OTB
+        penalty_k_range = np.round(np.arange(0.26, 0.45, 0.02), decimals=2)
+        lr_range = np.round(np.arange(0.3, 0.45, 0.02), decimals=2)
+        win_influ_range = np.round(np.arange(0.46, 0.55, 0.02), decimals=2)
+    elif args.dataset == 'uav':
+        # fine search on UVA
+        penalty_k_range = np.round(np.arange(0.01, 0.08, 0.02), decimals=2)
```

Comment:
put these arguments in `parse_args`, and set them as parameters that must be passed in.